src/3d/qgs3dmapsettings.cpp

```diff
@@ -513,7 +513,11 @@ void Qgs3DMapSettings::setOrigin( const QgsVector3D &origin )
 {
   QGIS_PROTECT_QOBJECT_THREAD_ACCESS
 
+  if ( origin == mOrigin )
```

Comment:
QgsVector3D== operator uses double == comparison. I think we should change that to use qgsDoubleNear so that this early-exit path has greater chance to being hit.